services/121-service/src/payments/reconciliation/excel/excel-reconciliation.service.ts

```diff
@@ -92,6 +95,22 @@ export class ExcelReconciliationService {
       countNotFound: number;
     };
   }> {
+    ////////////////////////////////////////////////////////////////////////////
+    // Preparing
+    ////////////////////////////////////////////////////////////////////////////
+    const program = await this.programRepository.findOne({
+      where: {
+        id: Equal(programId),
+      },
+      relations: ['programFspConfigurations'],
+    });
+    if (!program) {
+      throw new HttpException(
+        `No program with id ${programId} found`,
+        HttpStatus.NOT_FOUND,
+      );
+    }
```

Comment:
This http exception is not needed, because our authentication will throw an error if the program does not exist.

I am also okay with leaving this here for this PR, as it does not really hurt as I can throw it away later in the refactor.